integration_test/survey_details_screen_test.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/test_util.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  surveyDetailsScreenTest();
+}
+
+void surveyDetailsScreenTest() {
+  group('Survey Details Page', () {
+    // late Finder backButton;
```

Comment:
@manh-t Sorry for not mentioned on the PR description. I temporarily skip the UI Test for survey details since we have to test with go_router `extra`, I added the TODO here to comeback when having more spare time 😅 https://github.com/hoangnguyen92dn/survey-flutter-ic/blob/adff8a3fcb1718bd60192b7d7d2383fd45f4a761/integration_test/survey_details_screen_test.dart#L33

Reply:
@hoangnguyen92dn The newest version of `go_router` supports passing `extras` to `initialExtra` ([ref](https://github.com/flutter/flutter/issues/122780)). I think you might consider to upgrade go_router version 😄

Reply:
@hoangnguyen92dn so will you do the UI test in this PR, or you still prefer to skip the test and do it later? 🙏

Reply:
@manh-t I've just updated the test 🚀